File: Bindings~/java/src/main/java/com/alttester/Commands/InputActions/AltKeysDown.java
/*
    Copyright(C) 2026 Altom Consulting
*/

package com.alttester.Commands.InputActions;

import com.alttester.Commands.AltBaseCommand;
import com.alttester.IMessageHandler;

public class AltKeysDown extends AltBaseCommand {

  private AltKeysDownParams altKeysDownParameters;

  public AltKeysDown(IMessageHandler messageHandler, AltKeysDownParams altKeysDownParameters) {
    super(messageHandler);
    this.altKeysDownParameters = altKeysDownParameters;
  }

  public void Execute() {
    SendCommand(altKeysDownParameters);
    recvall(altKeysDownParameters, String.class);
  }
}

File: Assets/AltTester/Runtime/AltDriver/Commands/InputActions/AltKeysDown.cs
/*
    Copyright(C) 2024 Altom Consulting

    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.

    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with this program. If not, see <https://www.gnu.org/licenses/>.
*/

using System;

namespace AltTester.AltTesterUnitySDK.Driver.Commands
{
    public class AltKeysDown : AltBaseCommand
    {
        AltKeysDownParams cmdParams;

        public AltKeysDown(IDriverCommunication commHandler, AltKeyCode[] keyCodes, float power) : base(commHandler)
        {
            this.cmdParams = new AltKeysDownParams(keyCodes, power);
        }
        public void Execute()
        {
            CommHandler.Send(cmdParams);
            var data = CommHandler.Recvall<string>(cmdParams);
            ValidateResponse("Ok", data);
        }
    }
}
